types/lodash-fp/lodash-fp.utils.types.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import _ from 'lodash/fp'
+
+import { SafeCurry2, SafeCurry3 } from '../../src/libs/type-utils/lodash-fp-types'
+
+
+// Curry2
+const fnWith2ArgsFn1 = (_x: number, _y: string): string => {
+  return 'something interesting'
+}
+const fnWith2ArgsFn2 = (_x: number[], _y: string) : string => {
+  return 'something interesting'
+}
+
+const c2Fn1 = _.curry(fnWith2ArgsFn1) as SafeCurry2<typeof fnWith2ArgsFn1>
```

Comment:
The usage for `SafeCurryN` is pretty verbose and repetitive. Would it be possible to add a helper to simplify this? Like a `curry` function in lodash-fp-helpers.ts overloaded to return the appropriate `SafeCurryN` type?

Reply:
I love it!  I had a similar idea floating around in my head.  I will give it a go.

Reply:
Hmmm.... This could be tricky...  I am going to recommend this as a follow-up exercise to iterate on what is here.  I'd like to get this in to assist the TS folks, switch to another task, and loop back around in a follow-up PR

Reply:
Sounds good to me.